### ISSUE TITLE
Can not reproduce the results of the RVQ

### ISSUE THREAD

**Issue by buptxyb666** [EricGuo5513/momask-codes#43]
Thanks for your great work. 

As mentioned other issues, i also find the RVQ FID score(0.02 v.s. 0.06) can't be reproduced even following released opt.txt. I wonder if the results are related to the GPU platform or Pytorch version ?

**Comment by Murrol** [comment]
We found that it might relate to the version differences in packages when processing the dataset. We are working on it. Please directly email me for dataset inquiry.

**Comment by wang-zm18** [comment]
Thanks for the work! I have tried to repoduce the results of VRQ in Table 2 of the released paper, too.  The repoduced results are roughly the same except the FID metric in the Generartion setting on HumanMLD (0.232 vs  0.051). The code is `python eval_t2m_trans_res.py --res_name tres_nlayer8_ld384_ff1024_rvq6ns_cdp0.2_sw --dataset_name t2m --name t2m_nlayer8_nhead6_ld384_ff1024_cdp0.1_rvq6ns --gpu_id 1 --cond_scale 4 --time_steps 10 --ext evaluation`. I have not figured the problem out now. Thank you in advance!

**Comment by HitBadTrap** [comment]
> Thanks for your great work.
> 
> As mentioned other issues, i also find the RVQ FID score(0.02 v.s. 0.06) can't be reproduced even following released opt.txt. I wonder if the results are related to the GPU platform or Pytorch version ?

Have you solved the problem now? @buptxyb666 

Thanks.

**Comment by wang-zm18** [comment]
I also tried to retrained rvq on the HumanMLD dataset with 50 epochs, and got the similar FID. I  want to know how many epochs to train rvq on the HumanMLD dataset.  By the way, I found the mean (-0.86 ~1.46)and std (0.01~0.36)of HumanMLD  is small (compared to those of KIT-ML, maybe the unit is different), so the data input to the rvq distributes within a very large range (such as -350.52~576.75 in a batch). Is the normalization is right? Thank you in advance!

**Comment by Murrol** [comment]
Ideally, it should save the best checkpoint using exactly our scripts. While you can refer to the provided pretrained model to find in which epoch it’s saved. 
Yea, it could be related to some deviation from data. You can send me an email for the official preprocess dataset from HumanML3D project. The purpose of this normalization is to standardize the dataset distribution under an  assumption of Gaussian distribution. It could be some problems. But we just follow most of the benchmark settings.

**Comment by wang-zm18** [comment]
I have sent you the email and show my previous confusion and my updated opinion. The conclusion is the process is right. Thank you!

**Comment by yrio233** [comment]
Hello, thank you for your excellent work!
I am now trying to reproduce your VQ model and train my own vq model locally according to the opt.txt parameters you published, but there is still a gap in the FID indicator (as shown in the figure, 0.027 is the local training result). I would like to ask if it is a problem with the dataset or something else? This question has troubled me for a long time, and I look forward to your reply.

[our_vq_opt.txt](https://github.com/user-attachments/files/19445196/our_vq_opt.txt)

![Image](https://github.com/user-attachments/assets/5a82b0a8-4ee8-4e7a-a267-0b561bfbb769)
